File: JavaScript/7js 2/fusc.js
/*
|--------------------------------------------------------------------------
| The fusc function -- Part 1
|--------------------------------------------------------------------------
|
*/

function fusc(n) {
   return n == 0 || n == 1 ? n : n % 2 == 0 ? fusc(n / 2) : fusc((n - 1) / 2) + fusc((n + 1) / 2);
}


function fusc(n) {
   if (n == 0) return 0;
   if (n == 1) return 1;
   if (n % 2 == 0) return fusc(n / 2);
   if (n % 2 != 0) return fusc((n - 1) / 2) + fusc((n - 1) / 2 + 1);
}


function fusc(n) {
   return n < 2 ? n : fusc(n >> 1) + (n & 1) * fusc(++n >> 1);
}


function fusc(n) {
   return n < 2 ? n : fusc((n / 2) | 0) + (n & 1 ? fusc(((n / 2) | 0) + 1) : 0);
}


console.log(fusc(0), 0);
console.log(fusc(1), 1);
console.log(fusc(85), 21);

File: Python/7py 2/fusc.py
"""
|--------------------------------------------------------------------------
| The fusc function -- Part 1
|--------------------------------------------------------------------------
|
"""

def fusc(n):
   return n if n < 2 else fusc(n/2) if n % 2 == 0 else fusc((n-1)/2) + fusc((n+1)/2)


def fusc(n):
   if n == 0 : return 0
   if n == 1 : return 1
   if n%2==0 : return int(fusc(n/2))
   if n%2!=0 : return int(fusc((n-1)/2))+int(fusc((n-1)/2+1))


def fusc(n):
   return n if n < 2 else fusc(n // 2) + (n % 2 and fusc(n // 2 + 1))


def fusc(n):
   if n < 2: return n
   m, rem = divmod(n, 2)
   return fusc(m) + (rem and fusc(m+1))


print(fusc(0), 0) 
print(fusc(1), 1)
print(fusc(85), 21)